coderd/database/migrations/000186_oauth2_provider_codes.up.sql

```diff
@@ -0,0 +1,44 @@
+CREATE TABLE oauth2_provider_app_codes (
+    id uuid NOT NULL,
+    created_at timestamp with time zone NOT NULL,
+    expires_at timestamp with time zone NOT NULL,
+    hashed_secret bytea NOT NULL,
+    user_id uuid NOT NULL REFERENCES users (id) ON DELETE CASCADE,
+    app_id uuid NOT NULL REFERENCES oauth2_provider_apps (id) ON DELETE CASCADE,
+    PRIMARY KEY (id),
+    UNIQUE(app_id, hashed_secret)
+);
+
+COMMENT ON TABLE oauth2_provider_app_codes IS 'Codes are meant to be exchanged for access tokens.';
+
+CREATE TABLE oauth2_provider_app_tokens (
+    id uuid NOT NULL,
+    created_at timestamp with time zone NOT NULL,
+    expires_at timestamp with time zone NOT NULL,
+    hashed_secret bytea NOT NULL,
+    app_secret_id uuid NOT NULL REFERENCES oauth2_provider_app_secrets (id) ON DELETE CASCADE,
+    api_key_id text NOT NULL REFERENCES api_keys (id) ON DELETE CASCADE,
+    PRIMARY KEY (id),
+    UNIQUE(app_secret_id, hashed_secret)
+);
+
+COMMENT ON TABLE oauth2_provider_app_tokens IS 'Refresh tokens both provide a way to refresh an access tokens (API keys) and a way to link API keys with the OAuth2 app and secret that generated them.';
```

Comment:
Oooh because the hash of the token is the api key table right?

We should rename this `refresh_hash`. It's descriptive enough, we know it is a token.
And comment on the column:

```sql
COMMENT ON COLUMN oauth2_provider_app_tokens.refresh_hash IS 'Refresh tokens both provide a way to refresh an access tokens (API keys). An expired apikey can be refreshed if this app_token is not yet expired. Meaning this expiry can outlive an api key.';
```

Reply:
Or `hashed_refresh_token` and `hashed_code`, maybe.

Reply:
`refresh_hash` and the comment makes sense to me!  Will change.

Reply:
just something with `refresh`